packages/lib/ProtocolUtils.ts

```diff
@@ -0,0 +1,33 @@
+export function getNoteUrl(noteId: string) {
+	return `joplin://x-callback-url/openNote?id=${noteId}`;
+}
+
+export function getFolderUrl(folderId: string) {
+	return `joplin://x-callback-url/openFolder?id=${folderId}`;
+}
+
+export function getTagUrl(tagId: string) {
+	return `joplin://x-callback-url/openTag?id=${tagId}`;
+}
+
+export type Command = 'openNote' | 'openFolder' | 'openTag';
+
+export interface UlrInfo {
+    command: Command;
+    params: any;
+}
+
+export function parseUrl(s: string): UlrInfo {
+	if (!s.startsWith('joplin://')) return null;
```

Comment:
That should throw an error if the url is invalid. Also shouldn't it start with `joplin://x-callback-url/` ?

Reply:
fixed

Reply:
Actually I don't think throwing error here is a good idea because the error will not be visible in dev tools